core/.js/src/test/scala/io/udash/component/ListenableTest.scala

```diff
@@ -0,0 +1,123 @@
+package io.udash
+package component
+
+import io.udash.testing.UdashFrontendTest
+
+import scala.collection.mutable.ListBuffer
+
+private final class TestListenable extends Listenable {
+  override type EventType = TestListenableEvent
+
+  def fireEvent(): Unit = {
+    fire(TestListenableEvent(this))
+  }
+}
+private final case class TestListenableEvent(source: TestListenable) extends ListenableEvent
+
+class ListenableTest extends UdashFrontendTest {
+
+  def getEmptyRegistration(listenable: Listenable): Registration = {
```

Comment:
nit: could be a part of `TestListenable` implementation since `listenable` is always passed as the one and only argument